x-pack/plugins/event_log/server/es/cluster_client_adapter.ts

```diff
@@ -312,8 +296,256 @@ export class ClusterClientAdapter {
     }
   }
 
+  public async queryEventsForAlertInstancesSummaryAggregation(
+    index: string,
+    namespace: string | undefined,
+    ids: string[],
+    start?: string,
+    end?: string
+  ): Promise<
+    Array<{
+      alertId: string;
+      instances: AlertInstanceSummary[];
+      last_execution_state: Record<string, unknown>;
+    }>
+  > {
+    const namespaceQuery = getNamespaceQuery(namespace);
+    const esClient = await this.elasticsearchClientPromise;
+    const body = {
+      size: 0,
+      sort: { ['@timestamp']: { order: 'desc' } },
+      aggs: {
+        events: {
+          filter: {
```

Comment:
since there is only one top-level agg (`events`), I'm curious why the filter needs to go in that agg - couldn't it be at the top-level instead?  Filters at that level (perhaps the inner ones too, not sure) can get cached, which might end up being useful since if you're querying over all the alerts you can "see", you may well be doing more than one over a short period of time, and that would mean we might get some good cache hits.